lib/python/bind_data_access.h

```diff
@@ -449,9 +447,36 @@ void bind_data_properties(pybind11::class_<T, Ignored...> &c) {
       "shape",
       [](const T &self) {
         const auto &dims = self.dims();
```

Comment:
I don't know if this gets optimized by the compiler anyway, but you are requesting the `.sizes()` every time in the loop below. Could you just get the sizes here instead?
```suggestion
        const auto &sizes = self.dims().sizes();
```

Reply:
It probably will be optimised away, it only constructs a `span`. But I might as well store that instead of the dims.